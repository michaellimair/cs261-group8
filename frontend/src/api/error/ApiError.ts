class ApiError<T> extends Error {
  data?: T;

  constructor(message: string, data?: T) {
    super(message ?? 'An error has occurred in the API.');
    this.data = data;
  }

  static isApiError(obj: any): obj is ApiError<any> {
    return obj instanceof ApiError;
  }
}

export default ApiError;
