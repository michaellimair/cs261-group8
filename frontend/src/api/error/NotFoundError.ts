import ApiError from './ApiError';

class NotFoundError extends ApiError<never> {
  static defaultMessage: string = 'Not Found';

  constructor(message?: string) {
    super(message ?? NotFoundError.defaultMessage);
  }

  static isNotFoundError(obj: any): obj is NotFoundError {
    return obj instanceof NotFoundError;
  }
}

export default NotFoundError;
