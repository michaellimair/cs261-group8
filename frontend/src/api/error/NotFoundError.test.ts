import ApiError from './ApiError';
import NotFoundError from './NotFoundError';

describe('NotFoundError', () => {
  it('is instantiated successfully', () => {
    const errorData = {
      status_code: 404,
      message: 'An error has occurred',
      errors: [],
    };

    const err = new NotFoundError(errorData.message);

    expect(err.message).toBe(errorData.message);
    expect(err.data).toMatchObject(errorData);
    expect(err).toMatchSnapshot();
  });

  it('is a subclass of ApiError', () => {
    const err = new NotFoundError('err');

    expect(ApiError.isApiError(err)).toBe(true);
  });

  it('uses the default error message when instantiated without any arguments', () => {
    const err = new NotFoundError();

    expect(err.message).toBe(NotFoundError.defaultMessage);
  });

  describe('isNotFoundError', () => {
    it('identifies NotFoundError classes correctly', () => {
      const err = new NotFoundError('err');

      expect(NotFoundError.isNotFoundError(err)).toBe(true);
    });
  });
});
