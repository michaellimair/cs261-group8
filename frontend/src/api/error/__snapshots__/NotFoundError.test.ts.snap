// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundError is instantiated successfully 1`] = `[Error: An error has occurred]`;
